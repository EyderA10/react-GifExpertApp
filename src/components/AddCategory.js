import React, { useState } from 'react';
import PropTypes  from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setinputValue] = useState(''); // ''

    const handleInputChange = (e) => {
        const value = e.target.value;
        setinputValue(value);

    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setinputValue('');
        }

    }

    return (
        <div>
            <form onSubmit={handleSubmit}>
                <p>{inputValue}</p>
                <input
                    type="text"
                    value={inputValue}
                    onChange={handleInputChange}
                />
            </form>
        </div>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
