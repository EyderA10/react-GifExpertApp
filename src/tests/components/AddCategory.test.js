import React from 'react';
import AddCategory from '../../components/AddCategory';
const { shallow } = require("enzyme")


describe('Pruebas en addcategory', () => {

    const setCategories = jest.fn();
    let wrapeer;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapeer = shallow(<AddCategory setCategories= {setCategories}/>);

    })

    test('debe de mostrarse correctamente', () => {

        expect(wrapeer).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto', () => {
        const input = wrapeer.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {target: {value}});

        expect(wrapeer.find('p').text().trim()).toBe(value);
    })

    test('no debe de postear el onsubmit', () => {
        
        wrapeer.find('form').simulate('submit', { preventDefault(){}});
        
        expect(setCategories).not.toHaveBeenCalled();
        //toHaveBeenCalledTimes(2) se llama 2 veces
    })
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        //1. simular el inputchange
        const input = wrapeer.find('input');
        const value = 'Goku';
        input.simulate('change', {target: {value}});
        
        //2. simular el submit
        wrapeer.find('form').simulate('submit', {preventDefault(){}});
        //3. se debe de haber llamado el setCategories
        expect(setCategories).toHaveBeenCalled();
        //4.el valor del input debe de estar ''
        expect(input.prop('value')).toBe('');
        
    })
    
    
})
