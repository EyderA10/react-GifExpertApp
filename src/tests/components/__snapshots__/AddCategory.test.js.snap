// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en addcategory debe de mostrarse correctamente 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</div>
`;
